tests/test_course_published.py

```diff
@@ -0,0 +1,120 @@
+"""
+Tests for the course_published sinks.
+"""
+import logging
+from datetime import datetime
+from unittest.mock import patch
+
+import pytest
+import requests
+import responses
+from responses import matchers
+from responses.registries import OrderedRegistry
+
+from event_sink_clickhouse.sinks.course_published import CoursePublishedSink
+from event_sink_clickhouse.tasks import dump_course_to_clickhouse
+from test_utils.helpers import (
+    check_block_csv_matcher,
+    check_relationship_csv_matcher,
+    course_factory,
+    course_str_factory,
+    get_clickhouse_http_params,
+    mock_course_overview,
+    mock_detached_xblock_types,
+)
+
+
+@responses.activate(registry=OrderedRegistry)  # pylint: disable=unexpected-keyword-arg,no-value-for-parameter
+@patch("event_sink_clickhouse.sinks.course_published.CoursePublishedSink._get_detached_xblock_types")
+@patch("event_sink_clickhouse.sinks.course_published.CoursePublishedSink._get_modulestore")
+def test_course_publish_success(mock_modulestore, mock_detached, caplog):
```

Comment:
can we use the same pattern as in test_django_settings file? ie, create a test suite class